gcsfs/core.py

```diff
@@ -398,65 +562,77 @@ def rmdir(self, bucket):
             for v in self.dirs[''][:]:
                 if v['name'] == bucket:
                     self.dirs[''].remove(v)
-        self.dirs.pop(bucket, None)
```

Comment:
We do want to remove the entry from a cached bucket listing, no? Also, the references to `dirs` should be renamed anyway.

Reply:
Oh, I see - you expect invalidate, below, to do this. Still true about `dirs`.